src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -76,11 +77,11 @@ public boolean processNextMessage() throws ServiceBusException, InterruptedExcep
     private MessageProcessingResult processCreateCommand(IMessage message) {
         log.info("Started processing payment message with ID {}", message.getMessageId());
 
-        CreatePaymentMessage payment = null;
+        PaymentMessage payment = null;
 
         try {
             payment = paymentMessageParser.parse(message.getMessageBody());
-            paymentMessageHandler.handlePaymentMessage(payment);
+            paymentMessageHandler.handlePaymentMessage((CreatePaymentMessage)payment);
```

Comment:
Casting to CreatePaymentMessage is bad

Reply:
next PR.

Reply:
perhaps should apply interface across the codebase and only cast where it's needed ;). or some sort of way because `handlePaymentMesage` returns specific response only related to create event

Reply:
> perhaps should apply interface across the codebase and only cast where it's needed ;). or some sort of way because `handlePaymentMesage` returns specific response only related to create event

at the end  planning tot do sth like this #72